docs/development.md

```diff
@@ -0,0 +1,263 @@
+<!-- 
+
+@license BSD-3-Clause
+
+Copyright (c) 2019 Project Jupyter Contributors.
+Distributed under the terms of the 3-Clause BSD License.
+
+-->
+
+# Development
+
+> Development guide.
+
+## Introduction
+
+Before we begin, development requires some setup and configuration. What follows is an overview of environment requirements and a sequence of steps for getting up and running. We use [Git][git] for version control, and for most tasks, we use [npm][npm] scripts to help us get things done quickly and effectively. For the most part, the project is able to internally manage dependencies for testing and linting, so, once you follow the steps below, you should be ready to start developing!
+
+So, without further ado, let's get you started!
+
+## Prerequisites
+
+Development requires the following prerequisites:
+
+-   [Git][git]: version control
+-   [Python][python]: general purpose language (version `>= 3.6`)
+-   [pip][pip]: Python package manager (version `>= 9.0.0`)
+-   [Node.js][node-js]: JavaScript runtime (latest stable version is **strongly** recommended)
+-   [npm][npm]: package manager (version `> 2.7.0`)
+-   [JupyterLab][jupyterlab]: computational environment (version `>= 1.0.0`)
+
+While not required, you are encouraged to create an [Anaconda][anaconda] environment.
+
+```bash
+$ conda create -n jupyterlab-data-explorer -c conda-forge python=3.6 -c anaconda jupyterlab nodejs
```

Comment:
```suggestion
$ conda create -n jupyterlab-data-explorer -c conda-forge python=3.6 jupyterlab nodejs
```

Reply:
Okay. Question. I essentially took this from the Jupyter [docs](https://jupyterlab.readthedocs.io/en/stable/developer/extension_tutorial.html#install-nodejs-jupyterlab-etc-in-a-conda-environment). The `-c anaconda` is specifying that the remaining arguments be installed via the  Anaconda "channel", correct? If this is unnecessary, why do the Jupyter docs show otherwise (or maybe I am misreading them)?

Reply:
I believe you have to put all the channel names first. In the jupyter docs, it does this:

```
conda create -n jupyterlab-ext --override-channels --strict-channel-priority -c conda-forge -c anaconda jupyterlab cookiecutter nodejs git
```

I always prefer to just to use conda-forge and not the anaconda channel. I know we had a long discussion about this over slack at quansight just this week... @scopatz Do you have any recommendations on including anaconda channel or just conda forge in this command?

Reply:
Updated.

Reply:
yeah, I recommend conda-forge